src/protagonist/DLCS.AWS/ElasticTranscoder/ElasticTranscoderWrapper.cs

```diff
@@ -51,31 +51,52 @@ public class ElasticTranscoderWrapper : IElasticTranscoderWrapper
 
         return cache.GetOrAddAsync(presetLookupKey, async entry =>
         {
-            var presets = new Dictionary<string, string>();
-            var response = new ListPresetsResponse();
-                
-            do
-            {
-                var request = new ListPresetsRequest {PageToken = response.NextPageToken};
-                response = await elasticTranscoder.ListPresetsAsync(request, token);
+            var presets = await RetrievePresets(token, entry);
 
-                foreach (var preset in response.Presets)
-                {
-                    presets.Add(preset.Name, preset.Id);
-                }
+            var presetsDictionary = presets.Select(p => new { Key = p.Name, Value = p.Id })
+                .ToDictionary(pair => pair.Key, pair => pair.Value);
 
-            } while (response.NextPageToken != null);
+            return presetsDictionary;
+        }, cacheSettings.GetMemoryCacheOptions(CacheDuration.Long, priority: CacheItemPriority.Low));
+    }
 
-            if (presets.Count == 0)
-            {
-                logger.LogWarning("No ElasticTranscoder presets found");
-                entry.AbsoluteExpirationRelativeToNow = TimeSpan.FromSeconds(cacheSettings.GetTtl(CacheDuration.Short));
-            }
+    public Task<Preset?> GetPresetDetails(string name, CancellationToken token)

```

Comment:
think the cache should probably return a list, as that's what we're getting from AWS, and the key is technially a redundant value - think `GetPresetIdLookup` should return a dictionary though, to avoid potential (but highly unlikely) issues of multiple presets with the same name